Sources/LambdaConverters.Wpf/TemplateSelector.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Diagnostics;
+using System.Windows;
+using System.Windows.Controls;
+using JetBrains.Annotations;
+
+namespace LambdaConverters
+{
+    /// <summary>
+    /// A factory class used to create lambda-based instances of the <see cref="DataTemplateSelector"/> class.
+    /// </summary>
+    public static class TemplateSelector
+    {
+        sealed class Selector<I> : System.Windows.Controls.DataTemplateSelector
+        {
+            readonly Func<TemplateSelectorArgs<I>, DataTemplate> selectFunction;
+
+            SelectorErrorStrategy ErrorStrategy { get; }
+
+            bool IsSelectFunctionAvailable { get; }
```

Comment:
The private property `IsSelectFunctionAvailable` looks redundant because the same boolean value can always determined (very cheap) by comparing the `selectFunction` with null. Consider removing the property.